lib/Echidna/Processor.hs

```diff
@@ -41,155 +42,94 @@ instance Exception ProcException
+  , asserts :: M.HashMap ContractName [FunctionName]
+  , constantValues  :: M.HashMap ContractName (M.HashMap FunctionName [AbiValue])
+  , generationGraph :: M.HashMap ContractName (M.HashMap FunctionName [FunctionName])
+  } deriving (Show)
+
+instance FromJSON SlitherInfo where
+  parseJSON = withObject "slitherOutput" $ \o -> do
+    -- take the value under 'description' through the path - $['results]['printers'][0]['description']
+    description <- ((o .: "results" >>= (.: "printers")) <&> head) >>= (.: "description")
+    -- description is a JSON string, needs additional parsing
+    withEmbeddedJSON "descriptionString" parseDescription (String description)
+    where
+      parseDescription = withObject "description" $ \o -> do
+        payableFunctions <- o .: "payable"
+        constantFunctions <- o .: "constant_functions"
+        asserts <- o .: "assert"
+        constantValues'
+          -- the type annotation is needed
+          :: M.HashMap ContractName (M.HashMap FunctionName [[Maybe AbiValue]])
+          <- o .: "constants_used" >>= (traverse . traverse . traverse . traverse) parseConstant
+        -- flatten [[AbiValue]], the array probably shouldn't be nested, fix it in Slither
+        let constantValues = (fmap . fmap) (catMaybes . concat) constantValues'
+        generationGraph <-
+          o .: "functions_relations" >>= (traverse . traverse) (withObject "relations" (.: "impacts"))
+        pure SlitherInfo {..}
+
+      parseConstant :: Value -> Parser (Maybe AbiValue)
+      parseConstant = withObject "const" $ \o -> do
+        v <- o .: "value"
+        t <- o .: "type"
+        case t of
+          'u':'i':'n':'t':x ->
+            case AbiUInt <$> readMaybe x <*> readMaybe v of
+              Nothing -> failure v t
+              i -> pure i
+
+          'i':'n':'t':x ->
+            case AbiInt <$> readMaybe x <*> readMaybe v of
+              Nothing -> failure v t
+              i -> pure i
+
+          "string" ->
+            pure $ Just $ AbiString $ BSU.fromString v
+
+          "address" ->
+            case readMaybe v :: Maybe Int of
+              Just a ->
+                case readMaybe ("0x" ++ showHex a "") of
+                  Nothing -> failure v t
+                  a' -> pure $ AbiAddress <$> a'
+              Nothing -> failure v t
+
+          -- we don't need all the types for now
+          _ -> pure Nothing
+        where failure v t = fail $ "failed to parse constant: "++ t ++ " " ++ v
 
 -- Slither processing
-runSlither :: (MonadIO m, MonadThrow m) => FilePath -> [String] -> m [SlitherInfo]
-runSlither fp args = let args' = ["--ignore-compile", "--print", "echidna", "--json", "-"] ++ args in do
+runSlither :: (MonadIO m, MonadThrow m) => FilePath -> [String] -> m SlitherInfo
+runSlither fp extraArgs = do
   mp <- liftIO $ findExecutable "slither"
   case mp of
     Nothing -> throwM $ ProcessorNotFound "slither" "You should install it using 'pip3 install slither-analyzer --user'"
```

Comment:
we should probably just warn here rather than throw

if we want to require slither in the future, that's up for debae

Reply:
We discussed this a few weeks ago. We decide to make Slither mandatory for Echidna, since it is used to get the constants from the source code now. If you don't have Slither installed, Echidna won't be able to explore code like this one:

```
function f(uint x) {
  require(x == 424242);
  ...
}
``` 

Forcing the user to use Slither is not so bad, since it is easy to install.